---
title: "Creating Summary Statistics For Binding b/w miR and lncRNA and coding annotations"
output: html_notebook
---

# Load the necessary packages
```{r}
library(ggplot2)
library(scales)
```

Read in the depth per nucleotide file for each strand
```{r}
WTforward <- read.delim("./WT_depth_files/WTwatson_norRNA.depth", header=FALSE)
colnames(WTforward) = c("chrom","pos","depth")

WTreverse <- read.delim("./WT_depth_files/WTcrick_norRNA.depth", header=FALSE)
colnames(WTreverse) = c("chrom","pos","depth")

```

Read in lncRNA and other Bed File (for coding genes load in the no miR Beds)
```{r}
NONCODEv6_mm10.lncAndGene <- read.delim("./NONCODEv6_mm10.lncAndGene.bed", header=FALSE)
WholeGene_Bed = read.delim("./Coding_Gene_BEDs/NCBI_WholeGene_noMir.bed", header=FALSE)
Intron_Bed = read.delim("./Coding_Gene_BEDs/NCBI_Introns_noMir.bed", header=FALSE)
Exon_Bed = read.delim("./Coding_Gene_BEDs/NCBI_Exons_noMir.bed", header=FALSE)
UTR_Bed = read.delim("./Coding_Gene_BEDs/NCBI_3pUTR_noMir.bed", header=FALSE)

```

Find which entries in lncRNA BED file are Transcripts and which are genes
```{r}
NONCODEv6_mm10.lncAndGene$Is_Gene = grepl("NONMMUG",NONCODEv6_mm10.lncAndGene$V4)
```

load the miR genomic location BED file
```{r}
mmu_miR_bed = read.delim("./mmu_miR.bed", header = FALSE)
```

load the bed file of NONCODE entries that overlap with miR genomic loci
```{r}
NONCODE_miR_overlap_bed = read.delim("./NONCODE_miR_overlap.bed", header = FALSE)
```

label genes that contain microRNA genomic expression
```{r}
z = dim(NONCODEv6_mm10.lncAndGene)
for(i in 1:z[1]){
  NONCODEv6_mm10.lncAndGene$miR_gene[i] = (NONCODEv6_mm10.lncAndGene$V4[i] %in% NONCODE_miR_overlap_bed$V4)
  
  
}
```

slice the large NONCODE File into only genes and no miR genes
```{r}
NONCODE_slim = NONCODEv6_mm10.lncAndGene[NONCODEv6_mm10.lncAndGene$Is_Gene == TRUE,]
NONCODE_slim = NONCODE_slim[NONCODE_slim$miR_gene == FALSE,]
```


create temporary chunks for the forward strand
```{r}
Chrom1_temp_forward = WTforward[(WTforward$chrom %in% "chr1"),]
Chrom1_GL456211_random_temp_forward = WTforward[(WTforward$chrom %in% "chr1_GL456211_random"),]
Chrom1_GL456221_random_temp_forward = WTforward[(WTforward$chrom %in% "chr1_GL456221_random"),]
Chrom10_temp_forward = WTforward[(WTforward$chrom %in% "chr10"),]
Chrom11_temp_forward = WTforward[(WTforward$chrom %in% "chr11"),]
Chrom12_temp_forward = WTforward[(WTforward$chrom %in% "chr12"),]
Chrom13_temp_forward = WTforward[(WTforward$chrom %in% "chr13"),]
Chrom14_temp_forward = WTforward[(WTforward$chrom %in% "chr14"),]
Chrom15_temp_forward = WTforward[(WTforward$chrom %in% "chr15"),]
Chrom16_temp_forward = WTforward[(WTforward$chrom %in% "chr16"),]
Chrom17_temp_forward = WTforward[(WTforward$chrom %in% "chr17"),]
Chrom18_temp_forward = WTforward[(WTforward$chrom %in% "chr18"),]
Chrom19_temp_forward = WTforward[(WTforward$chrom %in% "chr19"),]
Chrom2_temp_forward = WTforward[(WTforward$chrom %in% "chr2"),]
Chrom3_temp_forward = WTforward[(WTforward$chrom %in% "chr3"),]
Chrom4_temp_forward = WTforward[(WTforward$chrom %in% "chr4"),]
Chrom4_GL456216_random_temp_forward = WTforward[(WTforward$chrom %in% "chr4_GL456216_random"),]
Chrom4_GL456350_random_temp_forward = WTforward[(WTforward$chrom %in% "chr4_GL456350_random"),]
Chrom4_JH584293_random_temp_forward = WTforward[(WTforward$chrom %in% "chr4_JH584293_random"),]
Chrom4_JH584295_random_temp_forward = WTforward[(WTforward$chrom %in% "chr4_JH584295_random"),]
Chrom5_temp_forward = WTforward[(WTforward$chrom %in% "chr5"),]
Chrom5_GL456354_random_temp_forward = WTforward[(WTforward$chrom %in% "chr5_GL456354_random"),] 
Chrom5_JH584297_random_temp_forward = WTforward[(WTforward$chrom %in% "chr5_JH584297_random"),] 
Chrom6_temp_forward = WTforward[(WTforward$chrom %in% "chr6"),] 
Chrom7_temp_forward = WTforward[(WTforward$chrom %in% "chr7"),] 
Chrom8_temp_forward = WTforward[(WTforward$chrom %in% "chr8"),] 
Chrom9_temp_forward = WTforward[(WTforward$chrom %in% "chr9"),] 
ChromM_temp_forward = WTforward[(WTforward$chrom %in% "chrM"),]  
ChromUn_GL456370_temp_forward = WTforward[(WTforward$chrom %in% "chrUn_GL456370"),] 
ChromUn_GL456387_temp_forward = WTforward[(WTforward$chrom %in% "chrUn_GL456387"),]  
ChromUn_JH584304_temp_forward = WTforward[(WTforward$chrom %in% "chrUn_JH584304"),]  
ChromX_temp_forward = WTforward[(WTforward$chrom %in% "chrX"),] 
ChromX_GL456233_random_temp_forward = WTforward[(WTforward$chrom %in% "chrX_GL456233_random"),] 
ChromY_temp_forward = WTforward[(WTforward$chrom %in% "chrY"),] 
```

create temporary chunks for the reverse strand
```{r}
Chrom1_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr1"),]
Chrom1_GL456211_random_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr1_GL456211_random"),]
Chrom1_GL456221_random_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr1_GL456221_random"),]
Chrom10_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr10"),]
Chrom11_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr11"),]
Chrom12_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr12"),]
Chrom13_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr13"),]
Chrom14_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr14"),]
Chrom15_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr15"),]
Chrom16_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr16"),]
Chrom17_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr17"),]
Chrom18_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr18"),]
Chrom19_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr19"),]
Chrom2_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr2"),]
Chrom3_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr3"),]
Chrom4_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr4"),]
Chrom4_GL456216_random_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr4_GL456216_random"),]
Chrom4_GL456350_random_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr4_GL456350_random"),]
Chrom4_JH584293_random_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr4_JH584293_random"),]
Chrom4_JH584295_random_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr4_JH584295_random"),]
Chrom5_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr5"),]
Chrom5_GL456354_random_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr5_GL456354_random"),] 
Chrom5_JH584297_random_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr5_JH584297_random"),] 
Chrom6_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr6"),] 
Chrom7_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr7"),] 
Chrom8_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr8"),] 
Chrom9_temp_reverse = WTreverse[(WTreverse$chrom %in% "chr9"),] 
ChromM_temp_reverse = WTreverse[(WTreverse$chrom %in% "chrM"),]  
ChromUn_GL456370_temp_reverse = WTreverse[(WTreverse$chrom %in% "chrUn_GL456370"),] 
ChromUn_GL456387_temp_reverse = WTreverse[(WTreverse$chrom %in% "chrUn_GL456387"),]  
ChromUn_JH584304_temp_reverse = WTreverse[(WTreverse$chrom %in% "chrUn_JH584304"),]  
ChromX_temp_reverse = WTreverse[(WTreverse$chrom %in% "chrX"),] 
ChromX_GL456233_random_temp_reverse = WTreverse[(WTreverse$chrom %in% "chrX_GL456233_random"),] 
ChromY_temp_reverse = WTreverse[(WTreverse$chrom %in% "chrY"),] 
```


treat the part below as a function to proccess each of the coding gene bed files, no need to save lncRNA_summary before re-using because lncRNA_slim is this data frame final

annotations done:
LNCSummary
WholeGene
Intron
exon

### start function
#change this each time
```{r}
NONCODE_slim = UTR_Bed
```


generate data table to populate summary binding for each transcript
```{r}
z = dim(NONCODE_slim)
lncRNA_num = z[1]
lncRNA_summary = data.frame(chrom = character(length = lncRNA_num), pos_start = numeric(length = lncRNA_num), pos_end = numeric(length = lncRNA_num), total_depth = numeric(length = lncRNA_num),geneid = character(length = lncRNA_num), depth_norm = numeric(length =lncRNA_num), 
                          stringsAsFactors = FALSE  )
```


Go through each annotation and find the appropriate temp depth file and sum reads for summary data frame
```{r}
for(i in 1:z[1]){
  if(NONCODE_slim$V1[i]=="chr1" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom1_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr1_GL456211_random" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom1_GL456211_random_temp_forward
  }

  if(NONCODE_slim$V1[i]== "chr1_GL456221_random" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom1_GL456221_random_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr10" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom10_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr11" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom11_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr12" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom12_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr13" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom13_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr14" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom14_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr15" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom15_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr16" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom16_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr17" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom17_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr18" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom18_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr19" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom19_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr2" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom2_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr3" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom3_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr4" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom4_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr4_GL456216_random" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom4_GL456216_random_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr4_GL456350_random" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom4_GL456350_random_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr4_JH584293_random" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom4_JH584293_random_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr4_JH584295_random" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom4_JH584295_random_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr5" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom5_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr5_GL456354_random" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom5_GL456354_random_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr5_JH584297_random" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom5_JH584297_random_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr6" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom6_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr7" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom7_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr8" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom8_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chr9" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = Chrom9_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chrM" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = ChromM_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chrUn_GL456370" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = ChromUn_GL456370_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chrUn_GL456387" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = ChromUn_GL456387_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chrUn_JH584304" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = ChromUn_JH584304_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chrX" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = ChromX_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chrX_GL456233_random" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = ChromX_GL456233_random_temp_forward
  }

  if(NONCODE_slim$V1[i]=="chrY" & NONCODE_slim$V6[i]=="+"){
    chrom_chunk_lncRNA_temp = ChromY_temp_forward
  }

  
    
  if(NONCODE_slim$V1[i]=="chr1" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom1_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr1_GL456211_random" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom1_GL456211_random_temp_reverse
  }

  if(NONCODE_slim$V1[i]== "chr1_GL456221_random" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom1_GL456221_random_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr10" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom10_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr11" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom11_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr12" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom12_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr13" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom13_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr14" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom14_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr15" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom15_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr16" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom16_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr17" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom17_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr18" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom18_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr19" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom19_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr2" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom2_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr3" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom3_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr4" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom4_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr4_GL456216_random" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom4_GL456216_random_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr4_GL456350_random" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom4_GL456350_random_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr4_JH584293_random" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom4_JH584293_random_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr4_JH584295_random" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom4_JH584295_random_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr5" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom5_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr5_GL456354_random" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom5_GL456354_random_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr5_JH584297_random" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom5_JH584297_random_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr6" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom6_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr7" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom7_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr8" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom8_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chr9" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = Chrom9_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chrM" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = ChromM_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chrUn_GL456370" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = ChromUn_GL456370_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chrUn_GL456387" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = ChromUn_GL456387_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chrUn_JH584304" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = ChromUn_JH584304_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chrX" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = ChromX_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chrX_GL456233_random" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = ChromX_GL456233_random_temp_reverse
  }

  if(NONCODE_slim$V1[i]=="chrY" & NONCODE_slim$V6[i]=="-"){
    chrom_chunk_lncRNA_temp = ChromY_temp_reverse
  }
   #create temp frame to merge into
   gene_temp_depth = data.frame(location = c(NONCODE_slim$V2[i]:NONCODE_slim$V3[i]))
   
   gene_temp_depth = merge(gene_temp_depth,chrom_chunk_lncRNA_temp,by.x = "location",by.y = "pos",all.x = TRUE)
   gene_temp_depth[is.na(gene_temp_depth)] = 0
   
   lncRNA_summary$chrom[i] = NONCODE_slim$V1[i]
   lncRNA_summary$pos_start[i] = NONCODE_slim$V2[i]
   lncRNA_summary$pos_end[i] = NONCODE_slim$V3[i]
   lncRNA_summary$total_depth[i] = sum(as.numeric(gene_temp_depth$depth))
   lncRNA_summary$geneid[i] = NONCODE_slim$V4[i]
   lncRNA_summary$depth_norm[i] = lncRNA_summary$total_depth[i]/(NONCODE_slim$V3[i]-NONCODE_slim$V2[i])
   
   
   message('\r',i,appendLF = FALSE)
   
  }

```
#Change here too
```{r}
UTR_summary = lncRNA_summary
```

##End Function

save everything
```{r}
saveRDS(WholeGene_summary, file = "./WholeGene_summary_directional_norRNA.rds")
saveRDS(Exon_summary, file = "./Exon_summary_directional_norRNA.rds")
saveRDS(Intron_summary, file = "./Intron_summary_directional_norRNA.rds")
saveRDS(UTR_summary, file = "./UTR_summary_directional_norRNA.rds")
saveRDS(LNC_summary, file = "./LNC_summary_directional_norRNA.rds")
```

read files back
```{r}
WholeGene_summary = readRDS(file = "./WholeGene_summary_directional_norRNA.rds")
UTR_summary = readRDS(file = "./UTR_summary_directional_norRNA.rds")
Exon_summary = readRDS(file = "./Exon_summary_directional_norRNA.rds")
Intron_summary = readRDS(file = "./Intron_summary_directional_norRNA.rds")
lncRNA_slim = readRDS(file = "./lncRNA_slim_directional_norRNA.rds")
```

assign the annotation to each summary file
```{r}
LNC_summary$annotation = "lncRNA"
UTR_summary$annotation = "UTR"
WholeGene_summary$annotation = "WholeGene"
Exon_summary$annotation = "Exon"
Intron_summary$annotation = "Intron"
```

Concatenate Them
```{r}
All_summary = rbind(LNC_summary, UTR_summary)
All_summary = rbind(All_summary, WholeGene_summary)
All_summary = rbind(All_summary, Exon_summary)
All_summary = rbind(All_summary, Intron_summary)
```



plot distribution of depths
```{r}
pdf(file = "./Whole_Transcript_Binding_hist_allAnntation_norRNA.pdf", width = 9, height = 6)
ggplot(data = All_summary, aes(x = total_depth, color = annotation)) + stat_bin(geom="step", position="identity", bins = 100)  + scale_color_manual(values  = c("Exon" = "maroon2", "Intron" = "maroon4", "lncRNA" = "black", "UTR" = "royalblue3", "WholeGene" = "royalblue4")) + scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),labels = trans_format("log10", math_format(10^.x))) + geom_vline(xintercept = 383447, color = "red") + ylab("Number of Annotations") + annotate(geom="label", x = 500000, y = 5000, label ="Malat1 #8 Highest Bound Gene \n Top 0.0091% of All Transcripts \n with >0 HITS-CLIP Reads", fill="white") + xlab("Total Transcipt Binding (reads)")
dev.off()
```








